packages/neon_framework/lib/src/cookie_store/utils.dart

```diff
@@ -0,0 +1,104 @@
+// ignore_for_file: constant_identifier_names
+
+import 'package:neon_framework/src/cookie_store/storable_cookie.dart';
+
+const int _DOT = 0x2E;
+const int _SLASH = 0x2F;
```

Comment:
This is the most efficient way to do a single character lookup.
Otherwise we'd have the use the `[]` or `String.characters` extension method creating lists just to be discarded again.

Reply:
Aren't you only using it for comparison in if-statements everywhere?

Reply:
yes but what does that change?

Reply:
I don't get why using the code points is so much more efficient than strings.

Reply:
Because `String.codeUnitAt(int index)` can directly lookup a single "character" while `String[]` will generate a list with every UTF-16 character in it just to discard the list again.

That's at least my guess and I've also seen the use of that pattern in other dart:core code (I did not do a performance analysis).
In the end it depends on the implementation of the executing dart environment. I'd guess that this doesn't make much difference in the vm while js probably has bigger losses.

Reply:
Ok makes sense, even though I doubt it will have any real world performance loss.